cros_gralloc/cros_gralloc_driver.cc

```diff
@@ -326,19 +318,32 @@ cros_gralloc_driver::cros_gralloc_driver()
 			DRV_INIT(drv_kms_, gpu_grp_type_, virtio_node_idx)
 		if (drv_kms_ && (virtio_node_idx != renderer_idx) && (drv_kms_ != drv_render_)) {
 			bool virtiopic_with_blob = drv_virtpci_with_blob(drv_kms_);
-			// The virtio pci device with blob feature could import buffers
-			// from i915, otherwise need use virtio to allocate scanout
-			// non-video buffers.
 			if (virtiopic_with_blob) {
+				//igpu SRIOV or dGPU SRIOV case
 				drv_logi("Virtio gpu device with blob\n");
 				if ((drv_kms_ != drv_render_) && drv_kms_)
 					DRV_DESTROY(drv_kms_)
 				drv_kms_ = drv_render_;
+			} else if (drv_is_dgpu(drv_video_) && drv_virtgpu_is_ivshm(drv_kms_)){
+				//is dgpu passthrough + ivshm case
+				drv_logi("dGPU with Virtio ivshm\n");
+				drv_ivshmem_ = drv_kms_;
+				drv_kms_ = drv_render_;
```

Comment:
What's so special about dGPU? 

The assignments here look a bit oblique. I see signs that this section of code is becoming confusing...